ts/parts/Interaction.ts

```diff
@@ -815,37 +844,49 @@ extend(Chart.prototype, /** @lends Chart.prototype */ {
                     ),
                     spill;
 
-                // If the new range spills over, either to the min or max,
-                // adjust the new range.
-                spill = paddedMin - newMin;
-                if (spill > 0) {
-                    newMax += spill;
-                    newMin = paddedMin;
-                }
-                spill = newMax - paddedMax;
-                if (spill > 0) {
-                    newMax = paddedMax;
-                    newMin -= spill;
-                }
+                // It is not necessary to calculate extremes on ordinal axis,
+                // because the are already calculated, so we don't want to
+                // override them.
+                if (!axis.options.ordinal) {
```

Comment:
`axisOpt` could be used since you have `axisOpt = axis.options,` in line No. 805